control-plane/stor-port/src/types/v0/store/volume.rs

```diff
@@ -428,9 +470,15 @@ impl SpecTransaction<VolumeOperation> for VolumeSpec {
 
     fn log_op(&self, operation: &VolumeOperation) -> (bool, bool) {
         match operation {
-            VolumeOperation::CreateSnapshot(_) | VolumeOperation::DestroySnapshot(_) => {
-                (false, false)
+            VolumeOperation::CreateSnapshot(_) => {
+                if !self.thin && !self.as_thin() {
```

Comment:
Is this check to be interpreted as `We are thick currently, and this is the first snapshot being created` ?

Reply:
Yes correct, do you think self.thin && !self.has_snapshots() would be clearer?

Reply:
Used has_snapshots check.